src/qibolab/instruments/emulator/pulse_simulator.py

```diff
@@ -180,8 +181,70 @@ def play(
             "output_states": output_states,
         }
 
+        results["demodulation"] = {}
+
+        if execution_parameters.acquisition_type is AcquisitionType.DISCRIMINATION:
```

Comment:
The point of the readout simulator is to simulate the transmission/phase change of the readout signal and the demodulation portion, so this should be `AcquisitionType.INTEGRATION`. Otherwise, we would just fallback onto the QuTiP state output for state discrimination.